CMakeLists.txt

```
@@ -0,0 +1,103 @@
+cmake_minimum_required(VERSION 3.22)
+
+include(FetchContent)
+
+project(ClpIrV1Decoder
+    LANGUAGES CXX C
+)
+
+# Enforce "Unix Makefiles" because "Ninja" does not work with the Emscripten toolchain on CXX_20 sources.
+if(NOT CMAKE_GENERATOR MATCHES "Unix Makefiles")
+    message(FATAL_ERROR "This project is intended to be built with Unix Makefiles on a Unix system.")
+endif()
```

nit: add space after `endif`
```suggestion
endif ()
```